src/oph/heratepalvelu/tep/tepSmsHandler.clj

```diff
@@ -95,16 +97,30 @@
     (log/info "Käsitellään" (count lahetettavat) "lähetettävää viestiä.")
     (when (seq lahetettavat)
       (doseq [nippu lahetettavat]
-        (when-not (= (:ei-niputettu c/kasittelytilat) (:kasittelytila nippu))
-          (if (c/has-time-to-answer? (:voimassaloppupvm nippu))
+        (log/info "Lähetetään SMS nipulle" nippu)
+        (if (= (:ei-niputettu c/kasittelytilat) (:kasittelytila nippu))
+          (log/error "Nipulla on käsittelytila ei-niputettu")
+          (if-not (c/has-time-to-answer? (:voimassaloppupvm nippu))
+            (try
+              (log/info "Vastausaika päättynyt" (:voimassaloppupvm nippu))
+              (tc/update-nippu nippu
+                               {:sms_lahetyspvm [:s (str (c/local-date-now))]
+                                :sms_kasittelytila
+                                [:s (:vastausaika-loppunut c/kasittelytilat)]})
+              (catch Exception e
+                (log/error "Virhe sms-lähetystilan päivityksessä nipulle,"
+                           "jonka vastausaika umpeutunut")
+                (log/error e)))
             (let [jaksot (tc/get-jaksot-for-nippu nippu)
                   oppilaitokset (c/get-oppilaitokset jaksot)
                   puhelinnumero (ohjaaja-puhnro nippu jaksot)
                   sms-kasittelytila (:sms_kasittelytila nippu)]
-              (when (and (some? puhelinnumero)
-                         (or (nil? sms-kasittelytila)
-                             (= sms-kasittelytila
-                                (:ei-lahetetty c/kasittelytilat))))
+              (if (or (nil? puhelinnumero)
+                      (and (some? sms-kasittelytila)
+                           (not= sms-kasittelytila
+                                 (:ei-lahetetty c/kasittelytilat))))
+                (log/warn "SMS:a ei voi lähettää, numero" puhelinnumero
```

Comment:
Tuli semmonen asia mieleen yhteys-/henkilötietoihin liittyen, että tämän HOKS-asetuksen mukaan me saadaan säilyttää yhteystietoja kolme kuukautta niiden käytön päättymisen jälkeen. Ja kun lokeja kuitenkin tullaan säilyttämään useita vuosia, niin pitäisikö yhteystietojen lokitusta välttää? Ne tietenkin voi auttaa ongelmanselvityksessä, mutta mikä olisi hyvä "riittävä" määrä herätettä yksilöivää tietoa lokirivillä?

Reply:
Hyvä pointti muuten. Tässä yhteydessä tuo on vähän haastavaa, kun nipun id itsessään pitää sisällään henkilötietoja. request-id taitaa (ohjaaja_ytunnus_kj_tutkinto + niputuspvm) ohella ainoa nipun yksilöivä tieto? :laughing:

Reply:
Tavoite on hyvä, mutta pitää tarkistaa, minkä pituiset on todelliset säilytysajat ja niiden vaatimukset.  Yleensä lokitietojen sensurointi ei ole oikea suunta, vaan lokitietojen lyhyempi säilytys.  Lokeja ei varmasti pitäisi säilyttää vuosia, vaan niin kauan kuin halutaan, että on selviteltävissä mitä tietyssä käsittelyssä on tapahtunut (esim just pari kuukautta); ja niin kauan kuin asian halutaan olevan selviteltävissä, myös yhteystietojen säilytykselle on peruste, nimittäin se, että muuten me ei tiedetä, kenen SMS-lähetys on epäonnistunut ja mikä se numero silloin oli.

Ehdotan, että tarkistetaan ensinnäkin, että 3kk aikaraja sisältää myös virheenselvityksen, ja sitten laitetaan lokien säilytysaika sen mukaiseksi.

Reply:
Hyvää pohdintaa! Katsotaan mihin suuntaan OPH:laiset haluaa lokitusta viedä pidemmällä tähtäimellä. Viedään kuitenkin tämä PR eteenpäin, niin ainakin on kättä pidempää kun pitää taas pian selvitellä jotain 😄

Reply:
(linkki  keskusteluun: https://opetushallitus.slack.com/archives/C03JN1DGD53/p1687420496763409)

Reply:
Tosiaan, tähän ei saatane vastausta nyt.  Ehdotan, että jatkettaisiin tämänhetkisellä sisällöllä ja tiketöitäisiin lokien säilytysajan lyhentäminen ellei jo ole.  Mulla on pahoja kokemuksia vanhoissa projekteissa siitä, että alettiin sensuroida lokeja (niistä tuli käytännössä ihan käyttökelvottomia).

Reply:
Joo, mennään tuolla sisällöllä :+1:

Reply:
https://jira.eduuni.fi/browse/EH-1520